### PR TITLE
f-braze-adapter@v4.0.0-beta.5 - adding in logging for better visibility

### PR DESCRIPTION
**Pull request by matthewhardern**
Adding updated logging to support a more refined logging for the adapter, while taking into account the new registry system, with logs now specifying the section of the adapter and a key indicating the consumer of the adapter. Tags have been added so that logs can be easily separated for display purposes.  variables are prefixed with logging so it obvious the keys are for logging purposes. I have also put a temp function in during the instantiation of the consumer class, because the dispatcher is not available during this time as the consumer has not been registered yet. I will look into a better solution at a later date.

### CODE REVIEW

File: packages/services/f-braze-adapter/src/services/BrazeConsumer.js [hunk 1]
```diff
@@ -32,9 +34,17 @@ class BrazeConsumer {
         callbacks = {},
         interceptInAppMessages = {},
         interceptInAppMessageClickEvents = {},
-        loggerCallbacks = {},
-        customFilters = []
+        logger = (type, message, data) => console.log(type, message, data),
+        customFilters = [],
+        key: $key
```
Comment (minrwhite) [inline packages/services/f-braze-adapter/src/services/BrazeConsumer.js]: Is it worth using `consumerKey` for this, given that there only seems to be one usage here, within a template string? The usage of `$key` looks a bit odd to me due to the recurrance of `$` in the tempate string, as well the potential confusion when mixing usage of `$` prefix for member variables.
```suggestion
        key: consumerKey
```

File: packages/services/f-braze-adapter/src/services/BrazeConsumer.js [hunk 2]
```diff
@@ -32,9 +34,17 @@ class BrazeConsumer {
...
     }) {
+        this.$logger = logger;
+        // key for logging
+        this.$consumerKey = `BrazeAdapter--consumer--${$key}`;
```
Comment (minrwhite) [inline packages/services/f-braze-adapter/src/services/BrazeConsumer.js]:
```suggestion
        this.$consumerKey = `BrazeAdapter--consumer--${consumerKey}`;
```

File: packages/services/f-braze-adapter/src/services/BrazeConsumer.js [hunk 3]
```diff
@@ -109,11 +141,10 @@ class BrazeConsumer {
 
     /**
      * Function that loops over all loggerCallbacks and calls them
```
Comment (minrwhite) [inline packages/services/f-braze-adapter/src/services/BrazeConsumer.js]: This description needs updating

**Review by minrwhite — COMMENTED** [review]
A couple of comments. It would be good to have test assertions that the log messages are sent to the event stream as expected in different scenarios, if this is possible. I'd understand if you'd prefer to do this in a later PR though.

File: packages/services/f-braze-adapter/src/services/BrazeConsumer.js (hunk 1 above)
Comment (matthewhardern) [reply packages/services/f-braze-adapter/src/services/BrazeConsumer.js]: Agreed see the confusion there, will make the change

**Comment by matthewhardern** [comment]
> A couple of comments. It would be good to have test assertions that the log messages are sent to the event stream as expected in different scenarios, if this is possible. I'd understand if you'd prefer to do this in a later PR though.

I agree It could do with some coverage to test the logging output occurs as expected. We do currently have coverage for the dispatcher in areas, so I will follow up in a later PR if thats ok to complete coverage for the logs.

**Review by minrwhite — APPROVED** [review]

**Review by minrwhite — APPROVED** [review]

File: packages/services/f-braze-adapter/CHANGELOG.md [hunk 4]
```diff
@@ -4,6 +4,14 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 
+v4.0.0-beta.5
+------------------------------
+*September 07, 2021*
+
+### Added
+- Finner detailed log events for greater error and event visibility
```
Comment (tommcclean) [inline packages/services/f-braze-adapter/CHANGELOG.md]: NIT
```suggestion
- Finer detailed log events for greater error and event visibility
```

**Pull request merged**